### PR TITLE
feat(cli): implement forest-cli f3 manifest

### LINKED ISSUE #4932: [F3] port `lotus f3 manifest` CLI command
# Summary

```
➜  lotus git:(master) ✗ ./lotus-calibnet f3 manifest -h
NAME:
   lotus f3 manifest - Gets the current manifest used by F3.

USAGE:
   lotus f3 manifest [command options] [arguments...]

OPTIONS:
   --output value  The output format. Supported formats: text, json (default: "text")
   --help, -h      show help
```

# Completion Criteria

- [ ]

# Additional Links & Resources

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

This PR ports `lotus f3 manifest` CLI command.

Changes introduced in this pull request:

- implement `forest-cli f3 manifest` command 

Output:
```
➜  forest git:(hm/forest-cli-f3-manifest) cargo run --bin forest-cli -- f3 manifest --output text
   Compiling forest-filecoin v0.21.1 (/home/me/git/forest)
    Finished `dev` profile [unoptimized] target(s) in 22.60s
     Running `target/debug/forest-cli f3 manifest --output text`
Manifest:
  Protocol Version:     4
  Paused:               false
  Initial Instance:     0
  Initial Power Table:  bafy2bzaceab236vmmb3n4q4tkvua2n4dphcbzzxerxuey3mot4g3cov5j3r2c
  Bootstrap Epoch:      2081674
  Network Name:         calibrationnet
  Ignore EC Power:      false
  Committee Lookback:   10
  Catch Up Alignment:   15s

  GPBFT Delta:                        6s
  GPBFT Delta BackOff Exponent:       2.0
  GPBFT Max Lookahead Rounds:         5
  GPBFT Rebroadcast Backoff Base:     6s
  GPBFT Rebroadcast Backoff Exponent: 1.3
  GPBFT Rebroadcast Backoff Spread:   0.1
  GPBFT Rebroadcast Backoff Max:      1m

  EC Period:            30s
  EC Finality:          900
  EC Delay Multiplier:  2.0
  EC Head Lookback:     0
  EC Finalize:          true

  Certificate Exchange Client Timeout:    10s
  Certificate Exchange Server Timeout:    1m
  Certificate Exchange Min Poll Interval: 30s
  Certificate Exchange Max Poll Interval: 2m
➜  forest git:(hm/forest-cli-f3-manifest) cargo run --bin forest-cli -- f3 manifest --output json
   Compiling forest-filecoin v0.21.1 (/home/me/git/forest)
    Finished `dev` profile [unoptimized] target(s) in 14.69s
     Running `target/debug/forest-cli f3 manifest --output json`
{
  "ProtocolVersion": 4,
  "Pause": false,
  "InitialInstance": 0,
  "BootstrapEpoch": 2081674,
  "NetworkName": "calibrationnet",
  "ExplicitPower": null,
  "IgnoreECPower": false,
  "InitialPowerTable": {
    "/": "bafy2bzaceab236vmmb3n4q4tkvua2n4dphcbzzxerxuey3mot4g3cov5j3r2c"
  },
  "CommitteeLookback": 10,
  "CatchUpAlignment": 15000000000,
  "Gpbft": {
    "Delta": 6000000000,
    "DeltaBackOffExponent": 2.0,
    "MaxLookaheadRounds": 5,
    "RebroadcastBackoffBase": 6000000000,
    "RebroadcastBackoffExponent": 1.3,
    "RebroadcastBackoffSpread": 0.1,
    "RebroadcastBackoffMax": 60000000000
  },
  "EC": {
    "Period": 30000000000,
    "Finality": 900,
    "DelayMultiplier": 2.0,
    "BaseDecisionBackoffTable": [
      1.3,
      1.69,
      2.2,
      2.86,
      3.71,
      4.83,
      6.27,
      7.5
    ],
    "HeadLookback": 0,
    "Finalize": true
  },
  "CertificateExchange": {
    "ClientRequestTimeout": 10000000000,
    "ServerRequestTimeout": 60000000000,
    "MinimumPollInterval": 30000000000,
    "MaximumPollInterval": 120000000000
  }
}
```

## Reference issue to close (if applicable)

Closes #4932

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/rpc/methods/f3/types.rs [hunk 1]
```diff
@@ -154,6 +154,82 @@ pub struct F3Instant {
...
+    pub minimum_poll_interval: Duration,
+    #[schemars(with = "u64")]
+    #[serde(with = "crate::lotus_json")]
+    pub maximum_poll_interval: Duration,
+}
+
+#[serde_as]
+#[derive(PartialEq, Debug, Clone, Serialize, Deserialize, JsonSchema)]
+#[serde(rename_all = "PascalCase")]
+pub struct F3Manifest {
```
Comment (hanabi1224) [inline src/rpc/methods/f3/types.rs:208]: @ruseinov These structs should be moved into `rust-f3` eventually

File: templates/cli/f3/manifest.stpl [hunk 2]
```diff
@@ -0,0 +1,29 @@
+Manifest:
```
Comment (hanabi1224) [inline templates/cli/f3/manifest.stpl:1]: Format comes from https://github.com/filecoin-project/lotus/blob/master/cli/templates/f3_manifest.go.tmpl

**Review by lemmih — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -48,6 +48,7 @@ directories = "5"
 displaydoc = "0.2"
 ethereum-types = "0.15"
 ez-jsonrpc-types = "0.3"
+fancy-duration = "0.9"
```
Comment (LesnyRumcajs) [inline Cargo.toml:51]: It seems like a pretty niche dependency without many users - do you think it makes sense depending on it or can we avoid? https://crates.io/crates/fancy-duration/reverse_dependencies

Comment (hanabi1224) [reply Cargo.toml:51]: Let me see if I can find a better replacement.  I first spotted https://crates.io/crates/human-duration but found it's not been updated for 3 years

Comment (hanabi1224) [reply Cargo.toml:51]: @LesnyRumcajs `fancy-duration` is the best I can find under https://crates.io/search?q=duration

`chrono::Duration` displays 15s as `PT15s` which follows https://en.wikipedia.org/wiki/ISO_8601#Durations but is not super readable for non-tech users

Comment (LesnyRumcajs) [reply Cargo.toml:51]: What about `humantime`? It's a bit of an abandonware, but we already have it in our dependencies.

**Comment by hanabi1224** [comment]
@LesnyRumcajs resolved a conflict in CHANGELOG, please re-approve, thanks!

**Pull request merged**